### PR TITLE
Add/ability to remove subscription if auto renewal is off

### PR DESCRIPTION
**Pull request by CodeyGuyDylan**
## Proposed Changes

This diff adds the ability to remove an Akismet purchase from your account

## Testing Instructions

1. Checkout this branch
2. Get your local environment running

** If you already have an Akismet purchase, you can ignore steps 3-6
3. Sandbox `public-api.wordpress.com`
4. Make sure you are using the store Sandbox
5. Go to `/checkout/akismet/ak_plus_yearly_1`
6. Complete the checkout

7. On the WPCOM backend, go to line 1595 in this file: fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Snqzva%2Qcyhtvaf%2Sjcpbz%2Qovyyvat%2Sfgber%2Qfhofpevcgvba.cuc%3Se%3Qs9rorpr0%231595-og and make that function return `false` immediately. This ensures the product won't be refunded and automatically removed from your account, because that's not what we are testing 😄 
8. Go to `/me/purchases` and select the Akismet Product
image
9. Turn Auto-Renew off, and choose any reason you'd like
image
image
10. Now you should see "Remove Subscription" instead of "Cancel Subscription" at the bottom
image
11. Click on it and remove the product. It should now be gone from the purchase management page
image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/lib/purchases/index.ts [hunk 1]
```diff
@@ -422,6 +422,16 @@ export function isCancelable( purchase: Purchase ) {
...
 
+/**
+ * Checks if a purchase is attached to a siteless holding site
+ *
+ * @param purchase The purchase to check
+ *
+ * @returns {boolean} True if the purchase is attached to a siteless holding site
+ */
+export function isSiteless( purchase: Purchase ): boolean {
+	return purchase.domain.includes( 'siteless' );
```
Comment (sirbrillig) [inline client/lib/purchases/index.ts]: I wonder if this is too general? What would happen if I purchased a domain registration for the domain `examplesiteless.com`?

Comment (CodeyGuyDylan) [reply client/lib/purchases/index.ts]: Hmmmm, that's a good point, I'll see if there's better data available to determine if it is a siteless site. I could just check specifically for `siteless.jetpack.com` or `siteless.akismet.com` if there isn't

Comment (CodeyGuyDylan) [reply client/lib/purchases/index.ts]: I found some existing functions that handle this, thank you for pointing this out

**Comment by CodeyGuyDylan** [comment]
Looks like this test is failing, which is strange because it works well for me locally. Will look into

**Comment by sirbrillig** [comment]
> Looks like this test is failing, which is strange because it works well for me locally. Will look into

🤔 The error is `Error: No QueryClient set, use QueryClientProvider to set one` which is referring to the changes made in https://github.com/Automattic/wp-calypso/pull/74676. I'm not sure why that's showing up here... maybe rebase the branch?

**Comment by CodeyGuyDylan** [comment]
> > Looks like this test is failing, which is strange because it works well for me locally. Will look into
> 
> 🤔 The error is `Error: No QueryClient set, use QueryClientProvider to set one` which is referring to the changes made in #74676. I'm not sure why that's showing up here... maybe rebase the branch?

In process, I don't know why my brain doesn't always immediately go to trying a rebase, it usually works 😅

File: client/me/purchases/manage-purchase/test/purchase-management-buttons.js [hunk 2]
```diff
@@ -164,4 +164,31 @@ describe( 'Purchase Management Buttons', () => {
...
+		const store = createMockReduxStoreForPurchase( {
+			...purchase,
+			domain: 'siteless.akismet.com',
+			product_id: 2311, // Akismet Plus Plan
+			product_slug: 'ak_plus_yearly_1',
+			auto_renew: 0,
+		} );
+
+		render(
+			<ReduxProvider store={ store }>
```
Comment (sirbrillig) [inline client/me/purchases/manage-purchase/test/purchase-management-buttons.js]: This just needs a `QueryClientProvider` as in the previous tests.

**Comment by CodeyGuyDylan** [comment]
Nah it still fails unfortunately, same reason, and only on the new test. No idea why because I can't reproduce it locally

File: client/me/purchases/manage-purchase/test/purchase-management-buttons.js (hunk 2 above)
Comment (CodeyGuyDylan) [reply client/me/purchases/manage-purchase/test/purchase-management-buttons.js]: Oh shoot, not sure how I skipped over that part 🤔 

I do still wonder why it kept passing for me locally, is this something TeamCity needs?

Comment (sirbrillig) [reply client/me/purchases/manage-purchase/test/purchase-management-buttons.js]: I think what happened is that it wasn't required until https://github.com/Automattic/wp-calypso/pull/74676, and this branch didn't include those changes, so it passed locally, but TC is clever and performs a merge with trunk before it runs tests. Since trunk did have those changes, the query client provider _was_ required in TC.

After the rebase, your new test should no longer pass locally and will also require the query client provider.

TLDR: not your fault. Just a conflict between branches. 😁

Comment (CodeyGuyDylan) [reply client/me/purchases/manage-purchase/test/purchase-management-buttons.js]: Interesting! Thanks for catching that 😄

Comment (CodeyGuyDylan) [reply client/me/purchases/manage-purchase/test/purchase-management-buttons.js]: > but TC is clever and performs a merge with trunk before it runs tests

I did not know this, very smart

File: client/me/purchases/remove-purchase/index.jsx [hunk 3]
```diff
@@ -194,6 +198,20 @@ class RemovePurchase extends Component {
...
+		} );
+
+		if (
+			isAkismetTemporarySitePurchase( purchase ) ||
+			isJetpackTemporarySitePurchase( purchase )
+		) {
+			successMessage = translate( '%(productName)s was removed from your account.', {
+				args: { productName },
+			} );
+		}
```
Comment (jboland88) [inline client/me/purchases/remove-purchase/index.jsx:213]: Nice improvement!!

**Review by jboland88 — APPROVED** [review]
This tested well and the code LGTM! 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7896549

Thank you @CodeyGuyDylan for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7896549).
